at_client/lib/src/util/at_client_validation.dart

```diff
@@ -1,37 +1,50 @@
 import 'package:at_client/at_client.dart';
+import 'package:at_client/src/util/network_util.dart';
 import 'package:at_commons/at_commons.dart';
+import 'package:at_utils/at_utils.dart';
 
 class AtClientValidation {
-  static void validateKey(String? key) {
-    if (key == null || key.isEmpty) {
-      throw AtKeyException('Key cannot be null or empty');
+  /// This function is used to validate the [key].
+  /// Throws [AtKeyException] if key has invalid values.
+  static bool _validateKey(String? key) {
+    /// Key cannot be null.
+    if (key == null) {
+      throw AtKeyException('AtKey is null');
     }
-    // Key cannot contain @
+
+    /// Key cannot be empty.
+    if (key.isEmpty) {
+      throw AtKeyException('Key cannot be empty');
+    }
+
+    /// Key cannot contain @
     if (key.contains('@')) {
       throw AtKeyException('Key cannot contain @');
     }
-    // Key cannot contain whitespaces
+
+    /// Key cannot contain whitespaces
     if (key.contains(' ')) {
       throw AtKeyException('Key cannot contain whitespaces');
     }
+    return true;
   }
 
   /// Validates the metadata of the key.
   /// Throws [AtKeyException] if metadata has invalid values.
-  static void validateMetadata(Metadata? metadata) {
+  static void _validateMetadata(Metadata? metadata) {
     if (metadata == null) {
       return;
-    }
+    } else
```

Comment:
Please remove else, if "IF" condition satisfies, exception is thrown or gets to next if condition.